spec/support/authentication_service_shared_examples.rb

```diff
@@ -0,0 +1,452 @@
+          new_default_service = FactoryGirl.build(service_sym, :default)
+          expect(new_default_service).to be_valid
+          non_default_service = FactoryGirl.build(service_sym)
+          expect(non_default_service).to be_valid
+        end
+      end
+    end
+  end
+
+  context 'get_user_for_access_token' do
+    context 'with valid token' do
+      context 'for first time user' do
+        it 'should return an unpersisted user with an unpersisted user_authentication_service assigned to user.current_user_authenticaiton_service' do
+          returned_user = subject.get_user_for_access_token(first_time_user_access_token)
+          expect(returned_user).not_to be_persisted
+          expect(returned_user.current_user_authenticaiton_service).not_to be_nil
+          expect(returned_user.current_user_authenticaiton_service).not_to be_persisted
+          expect(returned_user.current_user_authenticaiton_service.authentication_service_id).to eq(subject.id)
+        end
+      end
+
+      context 'for existing user' do
+        context 'already authenticated with this authentication service' do
+          it 'should return a persisted user with a persisted user_authentication_service assigned to user.current_user_authenticaiton_service' do
+            returned_user = subject.get_user_for_access_token(existing_user_access_token)
+            expect(returned_user).to be_persisted
+            expect(returned_user.id).to eq(existing_user.id)
+            expect(returned_user.current_user_authenticaiton_service).not_to be_nil
+            expect(returned_user.current_user_authenticaiton_service).to be_persisted
+            expect(returned_user.current_user_authenticaiton_service.id).to eq(existing_user_auth.id)
+          end
+        end
+
+        context 'not authenticated with this authentication service' do
+          it 'should return a persisted user with the an unpersisted user_authentication_service assigned to user.current_user_authenticaiton_service' do
+            returned_user = subject.get_user_for_access_token(existing_first_authenticating_access_token)
+            expect(returned_user).to be_persisted
+            expect(returned_user.id).to eq(existing_first_authenticating_user.id)
+            expect(returned_user.current_user_authenticaiton_service).not_to be_nil
+            expect(returned_user.current_user_authenticaiton_service).not_to be_persisted
+          end
+        end
+      end
+    end
+
+    context 'with invalid token' do
+      it {
+        expect {
+          subject.get_user_for_access_token(invalid_access_token)
+        }.to raise_error(InvalidAccessTokenException)
+      }
+    end
+  end
+end
+
+shared_examples 'an authentication request endpoint' do
+  # requires the following be let
+  #   first_time_user
+  #   first_time_user_access_token
+  #   existing_user
+  #   existing_user_access_token
+  #   invalid_access_token
+
+  it_behaves_like 'a GET request' do
```

Comment:
I envisioned the 'a GET request' shared example as a top level call in the main request spec files that wraps calls to more functional shared examples. I really should have documented it better.  :(